reviewer/app/GenotypePaths.cpp

```diff
@@ -235,18 +235,78 @@ getCandidateDiplotypePaths(int meanFragLen, const string& vcfPath, const LocusSp
         ++node;
     }
 
-    vector<Diplotype> pathsByDiplotype;
+    vector<Diplotype> diplotypes;
     const NodeId rightFlankNode = locusSpec.regionGraph().numNodes() - 1;
     const int rightFlankLength = locusSpec.regionGraph().nodeSeq(rightFlankNode).length();
     for (const auto& diplotypeNodes : nodesByDiplotype)
     {
-        Diplotype genotypePaths;
+        Diplotype diplotype;
         for (const auto& haplotypeNodes : diplotypeNodes)
         {
-            genotypePaths.emplace_back(&locusSpec.regionGraph(), 0, haplotypeNodes, rightFlankLength);
+            diplotype.emplace_back(&locusSpec.regionGraph(), 0, haplotypeNodes, rightFlankLength);
         }
-        pathsByDiplotype.push_back(genotypePaths);
+
+        if (diplotype.front() < diplotype.back())
+        {
+            std::iter_swap(diplotype.begin(), diplotype.end() - 1);
+        }
```

Comment:
What is going on here?

Reply:
This is done to consistently order haplotypes within a diplotype and hence to avoid a situation of having two diplotypes that differ only by the order of the constituent haplotypes.

For example `(LF)(CAG){4}(AT){5}(RF)/(LF)(CAG){2}(AT){3}(RF)` gets converted into `(LF)(CAG){2}(AT){3}(RF)/(LF)(CAG){4}(AT){5}(RF)`. (Paths are ordered using the corresponding node id tuples.) I've added a clarifying code comment.